### PR TITLE
Fix/nested-form-errors

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Change error extraction logic to correctly surface validation errors on nested fields.
Such as repeatable components on unique field validation

### Why is it needed?

Found in QA on the target branch
So the user can see the errors in their CM form

### How to test it?

Modified existing uniqueness E2E test to ensure correct errors are visible

### Related issue(s)/PR(s)

DX-1348

### CODE REVIEW

File: packages/core/admin/admin/src/components/Form.tsx [hunk 1]
```diff
@@ -669,12 +669,48 @@ const useField = <TValue = any,>(path: string): FieldValue<TValue | undefined> =
 
   const handleChange = useForm('useField', (state) => state.onChange);
 
-  const error = useForm('useField', (state) => getIn(state.errors, path));
+  const convertErrorFormat = (stateErrors: FormErrors<FormValues>) => {
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/components/Form.tsx:672]: `convertErrorFormat` to what? 😄

Comment (joshuaellis) [reply packages/core/admin/admin/src/components/Form.tsx:672]: on the flip side, this function code is really clear so well done 👏🏼

**Review by joshuaellis — COMMENTED** [review]
Do you think it's possible to add a few tests for the error message stuff? 😄 be nice for no one to break it and have our assumptions verified

File: packages/core/admin/admin/src/components/Form.tsx [hunk 2]
```diff
@@ -778,7 +820,7 @@ const Blocker = ({ onProceed = () => {}, onCancel = () => {} }: BlockerProps) =>
   return null;
 };
 
-export { Form, Blocker, useField, useForm, getYupValidationErrors };
+export { Form, Blocker, useField, useForm, getYupValidationErrors, FormProvider };
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/components/Form.tsx:823]: How confident are you that you're not exporting this provider from the admin package when you do this?

Comment (jhoward1994) [reply packages/core/admin/admin/src/components/Form.tsx:823]: Oops I see https://github.com/strapi/strapi/blob/0c5ac9b4a0965db69613ed48a167c99c6127d9e0/packages/core/admin/admin/src/index.ts#L15-L16

🤔 wdyt the best idea is? should we do named form exports from the package index?

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx [hunk 3]
```diff
@@ -69,7 +69,14 @@ const formatErrorMessages = (errors: FormErrors, parentKey: string, formatMessag
           )
         );
       } else {
-        messages.push(...formatErrorMessages(value, currentKey, formatMessage));
+        messages.push(
+          ...formatErrorMessages(
+            // @ts-expect-error TODO: check why value is not compatible with FormErrors
```
Comment (jhoward1994) [inline packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx:74]: @joshuaellis 

Providing 'initialerrors' to the Form seems like the best way to go. The only problem I'm having is with the FormError types here.

I've modified them to accept TranslationMessages but I can't work out why that has broken TS here in the CM. The type for values seems like it should be compatible.

(parameter) value: FormErrors<any> | TranslationMessage

any ideas?

Comment (joshuaellis) [reply packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx:74]: You could mark it as `todo`? it's probably just a bit of a bad recursive function 🤔

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**